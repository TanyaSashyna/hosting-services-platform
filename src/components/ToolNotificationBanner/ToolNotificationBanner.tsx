import React, { FC, useState, useEffect } from "react";
import { NotificationBanner, Link } from "@project-name-frontend/ui-package";

import { StatusTool } from "../../api/ToolApi";

type ToolNotificationBannerProps = {
  status: StatusTool;
  domain: string;
  dnsChangesRequired: boolean;
  openDnsChangesModal: (state: boolean) => void;
};

export const ToolNotificationBanner: FC<ToolNotificationBannerProps> = ({
  status,
  domain,
  dnsChangesRequired,
  openDnsChangesModal,
}) => {
  const [showBanner, setShowBanner] = useState<boolean>(false);

  useEffect(() => {
    if (!dnsChangesRequired && status === "enabled") {
      setShowBanner(true);
    }
  }, [dnsChangesRequired, status]);

  if (status === "enabled" && dnsChangesRequired) {
    return (
      <NotificationBanner
        icon="warning"
        intent="warning"
        title="Warning"
        body={
          <>
            Tool setup is complete, however, you need to{" "}
            <Link onClick={(): void => openDnsChangesModal(true)}>
              update your DNS records
            </Link>{" "}
            to fully activate your service. You’ll still be billed for Tool while
            DNS changes are pending.
          </>
        }
      />
    );
  }

  if (showBanner) {
    return (
      <NotificationBanner
        icon="check_circle"
        intent="positive"
        title={`Tool is active on ${domain}`}
        body="Your website’s DNS is now pointing to Tool."
        onClose={(): void => setShowBanner(false)}
      />
    );
  }

  return null;
};

ToolNotificationBanner.displayName = "ToolNotificationBanner";
