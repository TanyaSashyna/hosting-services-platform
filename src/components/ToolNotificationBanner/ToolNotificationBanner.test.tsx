import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { ToolNotificationBanner } from "./ToolNotificationBanner";

afterEach(cleanup);

describe("ToolNotificationBanner", () => {
  const mockFunc = jest.fn();

  it("renders ToolNotificationBanner and matches snapshot", () => {
    const { container, asFragment } = render(
      <ToolNotificationBanner
        status="enabled"
        domain="example.com"
        dnsChangesRequired={false}
        openDnsChangesModal={mockFunc}
      />
    );

    expect(container).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders when the status is enabled and open is true", () => {
    const { getByText } = render(
      <ToolNotificationBanner
        status="enabled"
        domain="example.com"
        dnsChangesRequired={false}
        openDnsChangesModal={mockFunc}
      />
    );

    expect(getByText("Tool is active on example.com")).toBeDefined();
  });

  it("close a Notification Banner", () => {
    const { getByText } = render(
      <ToolNotificationBanner
        status="enabled"
        domain="example.com"
        dnsChangesRequired={false}
        openDnsChangesModal={mockFunc}
      />
    );

    const closeBtn = getByText("close");
    fireEvent.click(closeBtn);

    expect(() => getByText("Tool is active on example.com")).toThrow(
      "Unable to find an element"
    );
  });

  it("renders when the status and dnsChangesRequired is true", () => {
    const { getByText } = render(
      <ToolNotificationBanner
        status="enabled"
        domain="example.com"
        dnsChangesRequired={true}
        openDnsChangesModal={mockFunc}
      />
    );

    expect(getByText("Warning")).toBeDefined();
  });

  it("trigger openDnsChangesModal function on DNS changes button click", () => {
    const { getByText } = render(
      <ToolNotificationBanner
        status="enabled"
        domain="example.com"
        dnsChangesRequired={true}
        openDnsChangesModal={mockFunc}
      />
    );

    const dnsChangesBtn = getByText("update your DNS records");
    fireEvent.click(dnsChangesBtn);

    expect(mockFunc).toHaveBeenCalled();
  });
});
