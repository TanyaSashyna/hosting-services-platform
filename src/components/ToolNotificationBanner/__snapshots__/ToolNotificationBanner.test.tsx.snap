// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolNotificationBanner renders ToolNotificationBanner and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="notificationBanner notificationBanner__border--positive"
  >
    <div
      class="notificationBanner__icon"
    >
      <i
        class="icon icon--positive"
      >
        check_circle
      </i>
    </div>
    <div
      class="notificationBanner__content"
    >
      <div
        class="notificationBanner__content__title"
      >
        Tool is active on example.com
      </div>
      <div
        class="notificationBanner__content__body"
      >
        Your websiteâ€™s DNS is now pointing to Tool.
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiIconButton-root iconButton iconButton--default notificationBanner__icon MuiIconButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <i
          class="icon"
        >
          close
        </i>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
